Source/ProgressionSystemRuntime/Public/Widgets/PSOverlayWidget.h

```diff
@@ -13,21 +14,56 @@ UCLASS()
 class PROGRESSIONSYSTEMRUNTIME_API UPSOverlayWidget : public UUserWidget
 {
 	GENERATED_BODY()
+
 public:
-	
+	/**
+	* Sets the visibility of the overlay elements and playing fade animation if needed 
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	* @param bShouldPlayFadeAnimation Defines if the fade animation should be played after or before widget is visible
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate, bool bShouldPlayFadeAnimation);
+
+protected:
+	/** overrides NativeTick to make the user widget tickable **/
+	virtual void NativeTick(const FGeometry& MyGeometry, float InDeltaTime) override;
+	/** Event to execute when widget is ready */
+	virtual void NativeConstruct() override;
+
+	/**
+	* Play the overlay elements fade-in/fade-out animation.
+	* Uses the internal FadeCurveFloatInternal initialized in NativeConstruct
+	*/
+	void TickPlayFadeOverlayAnimation();
+
 	/**
 	* Sets the visibility of the background overlay and lock icon.
 	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
 	*/
 	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
-	void SetOverlayVisibility(ESlateVisibility VisibilitySlate);
+	void SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate);
+
+	/** Overlay widget which is a root for all fade-in/out overlay elements */
+	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UOverlay> PSCOverlay = nullptr;
 	
-protected:
 	/** Background overlay for tint effect */
 	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
 	TObjectPtr<class UImage> PSCBackgroundOverlay = nullptr;
 
 	/** Background overlay lock icon */
 	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
 	TObjectPtr<class UImage> PSCBackgroundIconLock = nullptr;
```

Comment:
Please remove `PSCBackgroundOverlay` and `PSCBackgroundIconLock` as they should no longer be in the code. The decision to include a background image or icon is up to the UI designer. The only subwidget to keep is the root `PSCOverlay`.

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/20/commits/b31977e4e0c4f1a106d0c5d359e60d669bc2e155

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/20/commits/b31977e4e0c4f1a106d0c5d359e60d669bc2e155